### PR TITLE
SONARIAC-440 [CF, TF] FP on rule S6304: "Granting access to all resources" should not be raised on actions without resource-level permissions

### PR DESCRIPTION
**Pull request by nils-werner-sonarsource**

### CODE REVIEW

File: iac-common/src/main/java/org/sonar/iac/common/checks/ResourceAccessPolicyVector.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  private static final String VECTOR_FILE = "ResourceAccessPolicyVector.json";
+
+  private ResourceAccessPolicyVector() {
+  }
+
+  private static final List<String> RESOURCE_ACCESS_POLICIES = loadResourceAccessPolicies();
+
+  private static List<String> loadResourceAccessPolicies() {
+    try {
+      String resourceAccessPolicies = loadJsonFile(VECTOR_FILE);
```
Comment (mstachniuk) [inline iac-common/src/main/java/org/sonar/iac/common/checks/ResourceAccessPolicyVector.java:47]: Why not just call 
```
IOUtils.toString(VECTOR_FILE, StandardCharsets.UTF_8);
```
?

File: iac-common/src/test/java/org/sonar/iac/common/checks/CommonTestUtils.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  public static class TestTree extends AbstractTestTree {
+
+    public static Tree tree() {
+      return new TestTree();
+    }
+  }
+
+  public static class TestTextTree extends AbstractTestTree implements TextTree {
+
+    public static TextTree text(String value) {
```
Comment (mstachniuk) [inline iac-common/src/test/java/org/sonar/iac/common/checks/CommonTestUtils.java:48]: I would suggest renaming the method to `textTree` or similar.

File: iac-common/src/test/java/org/sonar/iac/common/checks/policy/PolicyValidatorTest.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+      properties(attribute("Action", "sns:Publish"), attribute("Resource", "*"), attribute("Effect", "Allow")),
+      properties(attribute("Action", "sns:Publish"), attribute("NotResource", "*"), attribute("Effect", "Deny")),
+      properties(attribute("Action", "foo:bar"), attribute("Resource", "*"), attribute("Effect", "Allow")),
+      properties(attribute("Action", "sns:Publish"), attribute("Effect", "Allow")),
+      properties(attribute("Action", "sns:Publish"), attribute("Resource", "*")),
+      properties(attribute("Action", "sns:Publish"), attribute("NotResource", "*"))
+    );
+
+    Policy policy = new Policy(null, tree -> statements);
+    assertThat(PolicyValidator.findInsecureStatements(policy)).hasSize(2);
```
Comment (mstachniuk) [inline iac-common/src/test/java/org/sonar/iac/common/checks/policy/PolicyValidatorTest.java:87]: I would suggest checking exactly what is in the result, not only the size.

File: iac-common/src/test/java/org/sonar/iac/common/checks/policy/PolicyValidatorTest.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+  }
+
+  static class TestIterable extends AbstractTestTree implements Iterable<Tree> {
```
Comment (mstachniuk) [inline iac-common/src/test/java/org/sonar/iac/common/checks/policy/PolicyValidatorTest.java:90]: Maybe it makes sense to move it to `CommonTestUtils`?

**Review by mstachniuk — APPROVED** [review]
LGTM, see my comments

File: iac-common/src/main/java/org/sonar/iac/common/checks/ResourceAccessPolicyVector.java (hunk 1 above)
Comment (nils-werner-sonarsource) [reply iac-common/src/main/java/org/sonar/iac/common/checks/ResourceAccessPolicyVector.java:47]: We don't want to relay on apache commons-io

**Pull request merged**